### PR TITLE
Organize 2D audio, camera, and physics in Viewport

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR organizes the 2D audio listener, camera, and physics methods in Viewport to be close to each other, and put them right above the 3D methods.

In addition to the grouping those audio/camera/physics together, this also moves the 2D audio listener code together, since previously the audio listener methods were scattered in 3 different places.

This PR takes a piece out of #66744, split off for easy reviewing.

This PR is the 2D version of #51474.

### CODE REVIEW

File: scene/main/viewport.h [hunk 1]
```diff
@@ -689,6 +664,27 @@ class Viewport : public Node {
...
+	Camera2D *get_camera_2d() const;
+	void _camera_2d_set(Camera2D *p_camera_2d);
+	void assign_next_enabled_camera_2d(const StringName &p_camera_group);
+
+	// Collider to frame
+	HashMap<ObjectID, uint64_t> physics_2d_mouseover;
+	// Collider & shape to frame
+	HashMap<Pair<ObjectID, int>, uint64_t, PairHash<ObjectID, int>> physics_2d_shape_mouseover;
+	// Cleans up colliders corresponding to old frames or all of them.
+	void _cleanup_mouseover_colliders(bool p_clean_all_frames, bool p_paused_only, uint64_t p_frame_reference = 0);
```
Comment (AThousandShips) [inline scene/main/viewport.h:686]: I don't think the private methods should be moved to public (in fact the 3D ones shouldn't be public in the first place either IMO)

Comment (aaronfranke) [reply scene/main/viewport.h:686]: Oh shoot, I forgot about visibility and did not realize the visibility was changing. I'll fix that ASAP, and fix the 3D ones being public, and also fix `_propagate_world_2d_changed` and `_validate_property` being public.

**Review by akien-mga — APPROVED** [review]
Seems good to me conceptually.

I'll let @AThousandShips confirm her code review before merging, as I didn't review in depth myself.

File: scene/main/viewport.h [hunk 2]
```diff
@@ -756,16 +768,11 @@ class Viewport : public Node {
 	void _own_world_3d_changed();
```
Comment (AThousandShips) [inline scene/main/viewport.h:768]: I'd move this and `world_3d` and `own_world_3d` as well to private

**Review by AThousandShips — APPROVED** [review]
LGTM, save for the comment, glanced over the moved code but don't expect any issues in them just being copied

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**